sdk/monitor/Azure.Monitor.Ingestion/tests/ConcurrencyCounterPolicy.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Monitor.Ingestion.Tests
+{
+    internal class ConcurrencyCounterPolicy : HttpPipelinePolicy
+    {
+        public volatile int Count;
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            Interlocked.Increment(ref Count);
```

Comment:
Should we try to add assertions directly in the Process/ProcessAsync methods to ensure the Count doesn't go above the max concurrency?

Reply:
Depending on if we are running in sequence or parallel the Asserts would have to be different. I thought it was easier to just do these checks in the different tests themselves

Reply:
Yes, you would have to take in the maxConcurrency as an argument to the policy's constructor. Checking in the tests will just do a one time snapshot whereas adding to the policy itself will ensure that the condition is always true.